spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jms/hornetq/HornetQConnectionFactoryConfiguration.java

```diff
@@ -39,7 +39,7 @@
 	public HornetQConnectionFactory jmsConnectionFactory(ListableBeanFactory beanFactory,
 			HornetQProperties properties) {
 		return new HornetQConnectionFactoryFactory(beanFactory, properties)
-				.createConnectionFactory(HornetQConnectionFactory.class);
+				.createConnectionFactory(HornetQSecuredConnectionFactory.class);
```

Comment:
I really don't like that because it looks like a secured connection is the default (while it isn't). It would be so much better if we would avoid to require custom constructors for the factory.

I had a look and don't see any other way though.